airbyte-cdk/python/airbyte_cdk/sources/declarative/auth/oauth.py

```diff
@@ -2,21 +2,64 @@
 # Copyright (c) 2022 Airbyte, Inc., all rights reserved.
 #
 
+from dataclasses import dataclass, field
 from typing import Any, List, Mapping, Optional, Union
 
 import pendulum
 from airbyte_cdk.sources.declarative.interpolation.interpolated_mapping import InterpolatedMapping
 from airbyte_cdk.sources.declarative.interpolation.interpolated_string import InterpolatedString
 from airbyte_cdk.sources.streams.http.requests_native_auth.abstract_oauth import AbstractOauth2Authenticator
+from dataclasses_jsonschema import JsonSchemaMixin
 
 
-class DeclarativeOauth2Authenticator(AbstractOauth2Authenticator):
+@dataclass
+class DeclarativeOauth2Authenticator(AbstractOauth2Authenticator, JsonSchemaMixin):
     """
     Generates OAuth2.0 access tokens from an OAuth2.0 refresh token and client credentials based on
     a declarative connector configuration file. Credentials can be defined explicitly or via interpolation
     at runtime. The generated access token is attached to each request via the Authorization header.
+
+    Attributes:
+        token_refresh_endpoint (Union[InterpolatedString, str]): The endpoint to refresh the access token
+        client_id (Union[InterpolatedString, str]): The client id
+        client_secret (Union[InterpolatedString, str]): Client secret
+        refresh_token (Union[InterpolatedString, str]): The token used to refresh the access token
+        config (Mapping[str, Any]): The user-provided configuration as specified by the source's spec
+        scopes (Optional[List[str]]): The scopes to request
+        token_expiry_date (Optional[Union[InterpolatedString, str]]): The access token expiration date
+        access_token_name (Union[InterpolatedString, str]): THe field to extract access token from in the response
+        expires_in_name (Union[InterpolatedString, str]): The field to extract expires_in from in the response
+        refresh_request_body (Optional[Mapping[str, Any]]): The request body to send in the refresh request
     """
 
+    token_refresh_endpoint: Union[InterpolatedString, str]
+    client_id: Union[InterpolatedString, str]
+    client_secret: Union[InterpolatedString, str]
+    refresh_token: Union[InterpolatedString, str]
+    access_token_name: Union[InterpolatedString, str]
+    expires_in_name: Union[InterpolatedString, str]
+    config: Mapping[str, Any] = field(default_factory=dict)
+    refresh_request_body: Optional[Mapping[str, Any]] = (None,)
+    scopes: Optional[List[str]] = None
+    refresh_token_expiry_date: Optional[Union[InterpolatedString, str]] = None  # this is a hack and point it out in the review
+    _token_expiry_date: pendulum.DateTime = field(init=False, repr=False)
+
+    def __post_init__(self, options: Mapping[str, Any]):
```

Comment:
This class is still a work in progress. The planned solution is to rename fields to avoid collisions with the retriever/eval methods. It should work, but is a pretty invasive solution that directly affects the language. I want to see if there are other workarounds.